blockly-ws-search/workspace_search.js

```diff
@@ -282,6 +293,9 @@ class WorkspaceSearch {
       Blockly.utils.dom.addClass(currPath, 'searchCurrent');
       this.updateCursor_(this.currentBlock_);
       this.scrollToVisible_(this.currentBlock_);
+      // Blockly.WidgetDiv.hide called in scroll is taking away focus.
+      // TODO: review setFocused call in Blockly.WidgetDiv.hide.
+      this.textInput_.focus();
```

Comment:
I had put this call inside of `next_` because I wanted to decouple the text input logic with the search behavior. Instead of moving it here, I'd like to suggest instead adding this logic to `scrollToVisible_` (and remove `this.textInput_.focus() `entirely):
```
const activeEl = document.activeElement;
this.workspace_.scroll(-targetLeft, -targetTop);
if (activeEl) {
  // Blockly.WidgetDiv.hide called in scroll is taking away focus.
  // TODO: review setFocused call in Blockly.WidgetDiv.hide.
  activeEl.focus();
}
```
I tested it locally and it appears to work the same. That way, where the focus goes back to is a bit more flexible and its moved closer to the `scroll` call that is culprit of taking the focus.

Reply:
updated